### ISSUE TITLE
Ability to access remote resources while developing locally🚀 Feature Request:

### ISSUE THREAD

**Issue by kyle-pena-nlp** [cloudflare/workers-sdk#4929]
### Describe the solution

Add an option to have wrangler execute **locally**, yet be able to fetch / interact with **remote resources** like KV and etc.?  

In my use case, I have some resources that are sensitive that I don't want to store on a dev box, even if they are just "dev keys".  So, I want my worker to be able to retrieve them via KV.  Yet, I want to be able to perform integration tests against locally hosted stuff - i.e.; i'd like to call fetch("http://127.0.0.1:80/getThingy").

**Comment by admah** [comment]
@kyle-pena-nlp have you tried `wrangler dev --remote`? Here's some more info on how to use it: https://developers.cloudflare.com/workers/observability/local-development-and-testing/#develop-locally-using-remote-resources-and-bindings

**Comment by kyle-pena-nlp** [comment]
> @kyle-pena-nlp have you tried `wrangler dev --remote`? Here's some more info on how to use it: https://developers.cloudflare.com/workers/observability/local-development-and-testing/#develop-locally-using-remote-resources-and-bindings

Certainly!  But isn't it the case that when developing with wranger dev --remote, that any requests issued through the fetch API will originate from cloudflare's servers rather than the local machine?  In which case, an integration tests involving a local server would be impossible without setting up a tunnel or something.

**Comment by mon** [comment]
As an additional note to justify this feature - debugging and profiling a worker via `wrangler dev --remote` is very painful.
- logs aren't as good as local (no colours, truncated outputs)
- taking a performance profile in devtools makes the worker extremely likely to hard crash under load (and take devtools and logs down with it)
- Parallel requests run very slowly and also crash sometimes
- `performance` counters only update on I/O due to Spectre mitigation, so it's difficult to profile CPU intensive work
- the RAM limit makes it harder to debug stream processing by pulling streams into RAM
- dev loop is slow because each bundle has to be uploaded per iteration

I'm probably going to make a second worker whose sole purpose is to be a `--remote` proxy for my local worker. I'm lucky in that it's just a protected REST service, and not R2/D1 which would need an API translation.

**Comment by ansonyuu** [comment]
+1! Would be super useful to have

**Comment by aa2kb** [comment]
Please add this

**Comment by dario-piotrowicz** [comment]
Hi all 😄 

I'm actually glad to let you know what we're working on this! and that we'll have something to share with you soon 😊 

There's no official ETA I can share, but my guess is that 3/4 weeks we should have something experimental you can try out 🙂

**Comment by dario-piotrowicz** [comment]
Hi again all 🙂👋 

As promised (3 weeks later! on the clock! 😁) we've got something you can try out! 😄 

Please have a look at our docs: https://developers.cloudflare.com/workers/development-testing/#remote-bindings

We also have a [discussion](https://github.com/cloudflare/workers-sdk/discussions/9660) where you can provide further questions or any feedback 🙂 

Given the above I am pretty confident that we can close this issue 🙂, if you have any more questions/feedback please comment on the discussion 😄
